models/base/scratch/snowplow_ecommerce_base_events_this_run.sql

```diff
@@ -0,0 +1,113 @@
+{% endif -%}
+
+{% if not var('snowplow__disable_ecommerce_checkouts', false) -%}
+    {% do contexts.append({'schema': var('snowplow__context_ecommerce_checkout_step'), 'prefix': 'checkout_', 'single_entity': True}) %}
+{% endif -%}
+
+{% if not var('snowplow__disable_ecommerce_page_context', false) -%}
+    {% do contexts.append({'schema': var('snowplow__context_ecommerce_page'), 'prefix': 'ecommerce_page_', 'single_entity': True}) %}
+{% endif -%}
+
+{% if not var('snowplow__disable_ecommerce_transactions', false) -%}
+    {% do contexts.append({'schema': var('snowplow__context_ecommerce_transaction'), 'prefix': 'transaction_', 'single_entity': True}) %}
+{% endif -%}
+
+{% if not var('snowplow__disable_ecommerce_carts', false) -%}
+    {% do contexts.append({'schema': var('snowplow__context_ecommerce_cart'), 'prefix': 'cart_', 'single_entity': True}) %}
+{% endif -%}
+
+
+{% if var('snowplow__enable_mobile_events') %}
+
+    {% do contexts.append({'schema': var('snowplow__context_mobile_session'), 'prefix': 'mob_session_', 'single_entity': True}) %}
+    {% do contexts.append({'schema': var('snowplow__context_screen'), 'prefix': 'mob_sc_view_', 'single_entity': True}) %}
+
+{% endif -%}
+
+
+{% set base_events_query = snowplow_utils.base_create_snowplow_events_this_run(
+                            sessions_this_run_table='snowplow_ecommerce_base_sessions_this_run',
+                            session_identifiers= session_identifiers(),
+                            session_sql=var('snowplow__session_sql', none),
+                            session_timestamp=var('snowplow__session_timestamp', 'collector_tstamp'),
+                            derived_tstamp_partitioned=var('snowplow__derived_tstamp_partitioned', true),
+                            days_late_allowed=var('snowplow__days_late_allowed', 3),
+                            max_session_days=var('snowplow__max_session_days', 3),
+                            app_ids=var('snowplow__app_id', []),
+                            snowplow_events_database=var('snowplow__database', target.database) if target.type not in ['databricks', 'spark'] else var('snowplow__databricks_catalog', 'hive_metastore') if target.type in ['databricks'] else var('snowplow__atomic_schema', 'atomic'),
+                            snowplow_events_schema=var('snowplow__atomic_schema', 'atomic'),
+                            snowplow_events_table=var('snowplow__events_table', 'events'),
+                            entities_or_sdes=contexts
+    ) %}
+
+{% set base_query_cols = get_column_schema_from_query( 'select * from (' + base_events_query +') a') %}
+
+with base_query as (
+    {{ base_events_query }}
+), prep as (
+    {% if target.type in ['postgres', 'redshift']%}
+        select
+            {% for col in base_query_cols | map(attribute='name') | list -%}
+                {% if col.lower() == 'session_identifier' -%}
```

Comment:
Does this come with a breaking change requiring a full refresh? I wonder if it would be better to align the naming to session and user identifier given that mobile events can also be processed and the unified model started to use this already. This is for a more general team discussion for the future but I think we should align the packages as much as possible (currently each names these two key fields differently, unfortunately). If the whole point of leaving them was to not introduce a breaking change I am fine to go ahead for now.

Reply:
Agreed, my opinion is that we are (I think) going to have a new ecommerce package soon anyway, so let's not mess around with changing loads of names - you only need a full refresh if you _change_ the identifier (which is the same in any package) so it's fine.